conformance/system_macros/make_field.ion

```diff
@@ -1,10 +1,68 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-// Test Cases:
-// make_field can be invoked using any type of macro reference.
-// the first argument must expand to a single non-null text value (`$0` is allowed)
-// annotations first argument are silently dropped
-// the second argument may be any single expression that expands to a single expression
-// make_field expands to a single unannotated struct containing only field(s) with the first argument as the field name,
-//     and the second argument value as the field value.
+(ion_1_1 "make_field can be invoked"
+         (each "in text with an unqualified macro name"
+               (text " (:make_field foo 0) ")
+               "in text with an unqualified macro address"
+               (text " (:22 foo 0) ")
+               "in text with a qualified macro name"
+               (text " (:$ion::make_field foo 0) ")
+               "in text using qualified system macro address 22"
+               (text " (:$ion::22 foo 0) ")
+               "in binary using system macro address 22"
+               (binary "EF 16 FB 66 6F 6F 60")
+               "in binary with a user macro address"
+               (binary "16 FB 66 6F 6F 60")
+               (produces {foo: 0} )))
+
+(ion_1_1 "the first argument"
+         (then "can be"
+               (each "a string"
+                     (text ''' (:make_field "foo" 0) ''')
+                     "a symbol with known text"
+                     (text "(:make_field foo 0)")
+                     "an expression that produces text"
+                     (text "(:make_field (:values 'foo') 0)")
+                     "annotated"
+                     (text "(:make_field bar::foo 0)")
+                     (produces {foo:0} ))
+               (then "a symbol with unknown text"
+                     (text "(:make_field $0 1)")
+                     // Could be (produces {$0:1} ), but some implementations don't support $0 nicely.
```

Comment:
Yeah, and it's also exposing a weakness in `IonElement`—it doesn't have any support for `$0`, but we're using it in the conformance runner in `ion-java` because it's so much easier to work with than `IonValue`.